### DISCUSSION TITLE
Links to ImageMagic documents result in File not found

### DISCUSSION THREAD

**Discussion by bmoffat-bp** [ImageMagick/ImageMagick#2587] [Help]
I went to the imagemagick.org site, and searched for "despeckle" and got a bunch of results, but all but the command line tools links (or at least the bunch I tried) end up with "File not found".  Is there a different place where I can read docs?

**Comment by snibgo** [comment]
How about: https://imagemagick.org/script/command-line-options.php#despeckle

**Reply by bmoffat-bp** [reply]
All that shows is "-despeckle   reduce the speckles within an image."  I have a client image that has a large white background that I want to remove, but there are a few dark-ish pixels that prevent -trim from removing a lot of the white background.  I was wondering if there was an option to "ignore random pixels from an otherwise uniform area".  I thought -despeckle might do the trick, but I tried it and it had no discernable effect.  Here's the command I used:
magick convert "$Image[0]" -despeckle -bordercolor white -border 0 -fuzz 1% -trim +repage -resize "$resize" -strip -profile "$ProfileDir\$profileCMYK" -profile "$ProfileDir\$profileSRGB" -quality $imageQuality $converted

**Reply by fmw42** [reply]
try -morphology close or -connected-components

see https://imagemagick.org/Usage/morphology/ and https://imagemagick.org/script/connected-components.php

Please always identify your IM version and platform and if possible post an example image.

**Reply by bmoffat-bp** [reply]
Version: ImageMagick 7.0.10-23 Q16 x64 2020-07-04 http://www.imagemagick.org
Copyright: Copyright (C) 1999-2018 ImageMagick Studio LLC
License: http://www.imagemagick.org/script/license.php
Visual C++: 192628806
Features: Cipher DPC HDRI OpenCL
Delegates (built-in): bzlib cairo flif freetype gslib heic jng jp2 jpeg lcms lqr lzma openexr pangocairo png ps raw rsvg tiff webp xml zlib
![sample](https://user-images.githubusercontent.com/71143187/93379641-1c950880-f82c-11ea-892f-ff7bf33a288f.jpg)

**Comment by matanshukry** [comment]
Also, you can change the host to prefix with `archive` instead of `www` for old links (such as forum questions).

So for the following google link for example, which results in file not found:
`http://www.imagemagick.org/discourse-server/viewtopic.php?t=32413`

you can change it to archive and reach it:
`http://archive.imagemagick.org/discourse-server/viewtopic.php?t=32413`

**Comment by fmw42** [comment]
If you just want to trim, then try

```
magick 93379641-1c950880-f82c-11ea-892f-ff7bf33a288f.jpg \
\( -clone 0 -negate -threshold 10% -morphology close octagon:2 -morphology open octagon:3 \) \
-alpha off -compose copy_opacity -composite -compose over \
-background white -flatten -trim +repage result0.png
```

![result0](https://user-images.githubusercontent.com/13458218/93395683-2d914a00-f82b-11ea-82b3-7b17d871327d.png)

If you want to make the background transparent, try one of the following:


```
magick 93379641-1c950880-f82c-11ea-892f-ff7bf33a288f.jpg \
\( -clone 0 -negate -threshold 3% -morphology close octagon:2 -morphology open octagon:3 \) \
-alpha off -compose copy_opacity -composite result1.png
```

![result1](https://user-images.githubusercontent.com/13458218/93395729-4568ce00-f82b-11ea-93de-e11ae8f2133f.png)


```
magick 93379641-1c950880-f82c-11ea-892f-ff7bf33a288f.jpg \
\( -clone 0 -negate -threshold 3% -type bilevel \
-define connected-components:area-threshold=5000 \
-define connected-components:mean-color=true \
-connected-components 8 \) \
-alpha off -compose copy_opacity -composite result2.png
```

![result2](https://user-images.githubusercontent.com/13458218/93395741-4ac61880-f82b-11ea-84f9-9abdaaac20ba.png)

**Reply by bmoffat-bp** [reply]
First of all, I would like to just say how appreciative I am of how quickly you all are responding.  Seriously - Thank you!

That first transform goes too far, but the others look promising!

**Reply by fmw42** [reply]
The first one is just a trim of the background, so the size of the image is reduced in dimensions. The other two do not trim, they simply make the existing background transparent. The size of the image is unchanged.
